llvm/lib/ProfileData/InstrProfWriter.cpp

```diff
@@ -470,17 +470,19 @@ static uint64_t writeMemProfRecords(
 // Serialize MemProfFrameData.  Return FrameTableOffset.
 static uint64_t writeMemProfFrames(
     ProfOStream &OS,
-    llvm::MapVector<memprof::FrameId, memprof::Frame> &MemProfFrameData) {
+    llvm::MapVector<memprof::FrameId, memprof::Frame> &MemProfFrameData,
+    memprof::IndexedVersion Version) {
+  memprof::FrameWriterTrait FrameWriter(Version);
   OnDiskChainedHashTableGenerator<memprof::FrameWriterTrait>
       FrameTableGenerator;
   for (auto &[FrameId, Frame] : MemProfFrameData) {
     // Insert the key (frame id) and value (frame contents).
-    FrameTableGenerator.insert(FrameId, Frame);
+    FrameTableGenerator.insert(FrameId, Frame, FrameWriter);
```

Comment:
Are there changes needed in this patch to the insert() and Emit() methods of OnDiskChainedHashTableGenerator to take the writer trait?

Reply:
> Are there changes needed in this patch to the insert() and Emit() methods of OnDiskChainedHashTableGenerator to take the writer trait?

Yes, `FrameWriter` carries the version, so we need to pass `FrameWriter` to both `insert` and `Emit`.

Reply:
Ok, I think that part is missing from the PR?